Source/Core/DolphinQt/DiscordJoinRequestDialog.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#ifdef USE_DISCORD_PRESENCE
+
+#include <QGridLayout>
+#include <QLabel>
+#include <QPixmap>
+#include <QPushButton>
+
+#include <discord-rpc/include/discord_rpc.h>
+
+#include "Common/HttpRequest.h"
+#include "Common/StringUtil.h"
+
+#include "DolphinQt/DiscordJoinRequestDialog.h"
+
+DiscordJoinRequestDialog::DiscordJoinRequestDialog(QWidget* parent, const char* id,
+                                                   const std::string& discord_tag,
+                                                   const std::string& avatar)
+    : QDialog(parent), m_user_id(id), m_close_timestamp(std::time(nullptr) + s_max_lifetime_seconds)
+{
+  setWindowTitle(tr("Request to Join Your Party"));
+  setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+
+  QPixmap avatar_pixmap;
+
+  if (avatar.empty())
+  {
+    const std::string avatar_endpoint =
+        StringFromFormat("https://cdn.discordapp.com/avatars/%s/%s.png", id, avatar.c_str());
+
+    Common::HttpRequest request;
+    Common::HttpRequest::Response response = request.Get(avatar_endpoint);
+
+    if (response.has_value())
+      avatar_pixmap.loadFromData(response->data(), static_cast<uint>(response->size()), "png");
+  }
+
+  CreateLayout(discord_tag, avatar_pixmap);
+  ConnectWidgets();
+}
+
+std::time_t DiscordJoinRequestDialog::GetCloseTimestamp() const
+{
+  return m_close_timestamp;
+}
+
+void DiscordJoinRequestDialog::CreateLayout(const std::string& discord_tag, const QPixmap& avatar)
+{
+  m_main_layout = new QGridLayout;
+
+  QString invite_button_text = QString::fromWCharArray(L"\u2714 ");
```

Comment:
These variables seem unnecessary. You should be able to concatenate within the QPushButton argument, though you shouldn't need to concatenate at all as it shouldn't be necessary to prefix these as wide char and UTF-32, standard char  array can handle Unicode just fine.